### PR TITLE
AEA-1542: Prescribing Refactor Branch

### PR DESCRIPTION
**Pull request by callumtipper-IW**
…tionerRole

### CODE REVIEW

File: models/fhir/common.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export interface Reference<T extends Resource> {
 
 export function createReference<T extends Resource>(reference: string): Reference<T> {
   return {
-    reference: getFullUrl(reference)
+    reference: getFullUrl(reference),
```
Comment (dan-jones-iw) [inline models/fhir/common.ts:61]: why

**Review by dan-jones-iw — APPROVED** [review]

File: coordinator/src/services/translation/response/practitioner-role.ts [hunk 2]
```diff
@@ -1,9 +1,10 @@
-import {convertName, convertTelecom, generateResourceId, joinArrayWithSpaces} from "./common"
+import {convertName, convertTelecom, generateResourceId, humanNameArrayToString} from "./common"
 import {toArray} from "../common"
 import {fhir, hl7V3} from "@models"
 import {createPractitionerOrRoleIdentifier} from "./identifiers"
 import {createPractitionerIdentifier} from "./practitioner"
 import {getOrganizationCodeIdentifier} from "./organization"
+import {createIdentifierReference} from "../../../../../models/fhir"
```
Comment (dan-jones-iw) [inline coordinator/src/services/translation/response/practitioner-role.ts:7]: update import

File: coordinator/src/services/translation/response/practitioner-role.ts [hunk 3]
```diff
@@ -43,6 +45,18 @@ export function createRefactoredPractitionerRole(
     code: createJobRoleNameCode(hl7AgentPerson.code._attributes.code),
     telecom: toArray(hl7AgentPerson.telecom)[0]?._attributes ? convertTelecom(hl7AgentPerson.telecom) : undefined
   }
+
+  const healthCareProviderLicense = representedOrganization.healthCareProviderLicense
+  if (healthCareProviderLicense) {
```
Comment (dan-jones-iw) [inline coordinator/src/services/translation/response/practitioner-role.ts:50]: what to do if this isnt present in v3 message? do we need to throw?

File: coordinator/src/services/translation/common/index.ts [hunk 4]
```diff
@@ -76,6 +77,67 @@ export function resolveReference<T extends fhir.Resource>(bundle: fhir.Bundle, r
...
+  if (isReference(organization)) {
+    return resolveReference(bundle, organization)
+  } else {
+    const location = practitionerRole.location
+      .map(location => resolveReference(bundle, location))
+      .map(location => location.address)
+    return {
+      resourceType: "Organization",
+      identifier: [organization.identifier],
+      //Will break for Primary Care
```
Comment (booshibooshi) [inline coordinator/src/services/translation/common/index.ts:105]: do we want to throw a not-supported or something for primary-care?

File: coordinator/src/services/translation/request/practitioner.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ import {
   getIdentifierValueOrNullForSystem,
   identifyMessageType,
   onlyElement,
-  onlyElementOrNull,
+  onlyElementOrNull, resolveHealthcareService, resolveOrganization, resolvePractitioner,
```
Comment (booshibooshi) [inline coordinator/src/services/translation/request/practitioner.ts:9]: 😢

File: coordinator/src/services/translation/common/index.ts (hunk 4 above)
Comment (booshibooshi) [reply coordinator/src/services/translation/common/index.ts:105]: discussed, raised ticket to cover the fix instead of flagging here

**Review by booshibooshi — APPROVED** [review]

**Review by booshibooshi — DISMISSED** [review]

**Review by booshibooshi — APPROVED** [review]

**Pull request merged**